.github/workflows/ci.yml

```diff
@@ -0,0 +1,755 @@
+name: Continuous Integration
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+#### Global environment variables
+env:
+  DYNAMODB_HOSTNAME: dynamodb
+  MYSQL_HOSTNAME: mysql
+  MYSQL_DATABASE: testdb
+  MYSQL_USERNAME: root
+  MYSQL_PASSWORD: Password123
+  DISABLE_TESTCONTAINERS: true
+  AWS_ACCESS_KEY_ID: dummy_key
+  AWS_SECRET_ACCESS_KEY: dummy_secret
+  AWS_DEFAULT_REGION: us-west-2
+  AWS_REGION: us-west-2
+  GO111MODULE: "on"
+
+
+jobs:
+  #### Java
+  java-secure-memory:
+    name: Build Java Secure Memory
+    runs-on: ubuntu-latest
+    container:
+      image: adoptopenjdk/maven-openjdk11
+      options: --ulimit core=-1 --ulimit memlock=-1:-1
+    steps:
+    - name: Checkout the repository
+      uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+    - name: Cache Java packages
+      uses: actions/cache@1a9e2138d905efd099035b49d8b7a3888c653ca8
+      with:
+        path: ~/.m2
+        key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: ${{ runner.os }}-m2
+    - name: Build
+      run: |
+        cd java/secure-memory
+        ./scripts/clean.sh
+        ./scripts/build.sh
+    - name: Test
+      run: |
+        cd java/secure-memory
+        ./scripts/test.sh
+
+  release-java-secure-memory:
+    if: contains( github.ref, 'release' )
+    name: Release Java Secure Memory
+    needs: java-secure-memory
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+      - name: Fetch all tags
+        run: git fetch --prune --unshallow --tags
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@8764a52df183aa0ccea74521dfd9d506ffc7a19a
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }} # Value of the GPG private key to import
+          gpg-passphrase: MAVEN_GPG_PASSPHRASE # env variable for GPG private key passphrase
```

Comment:
Why is the gpg private key using the `${{ }}` syntax, while the rest are just variable names? Should/can these all be done in the same manner?

Reply:
Also, let's ensure that all of these variables are populated in the repo secrets before merging.

Reply:
I followed the instructions from this issue: https://github.com/actions/setup-java/issues/83#issuecomment-825146155.
I could test if it works with an all `${{ }}` syntax.